service/matching/taskListManager_test.go

```diff
@@ -191,7 +191,7 @@ func TestDescribeTaskList(t *testing.T) {
 	require.NotNil(t, taskListStatus)
 	require.Zero(t, taskListStatus.GetAckLevel())
 	require.Equal(t, taskCount, taskListStatus.GetReadLevel())
-	require.Equal(t, taskCount, taskListStatus.GetBacklogCountHint())
+	require.Equal(t, int64(0), taskListStatus.GetBacklogCountHint())
```

Comment:
Not sure I understand. In the test 3 tasks are pushed to the tasklist. Why a backlog is still 0? Or are they immediately processed?
I suggest adding a test that verifies the backlog count/metric emition with tally.TestScope.

Reply:
The original value is in-memory backlog, and the new value is the in-db backlog. The test generates the backlog by changing the in-memory backlog, so after the change, the value is changed.